src/commands/Moderation/ban.ts

```diff
@@ -23,10 +18,13 @@ export default class extends ModerationCommand {
 		return message.guild!.settings.get(GuildSettings.Events.BanAdd) ? { unlock: message.guild!.moderation.createLock() } : null;
 	}
 
-	public async handle(message: KlasaMessage, target: User, member: GuildMember, reason: string, _prehandled: Unlock, duration: number | null) {
-		await message.guild!.members.ban(target.id, { days: Number(message.flagArgs.day || message.flagArgs.days) || 0, reason });
-
-		return this.sendModlog(message, target, reason, null, duration);
+	public handle(message: KlasaMessage, target: User, reason: string | null, duration: number | null) {
+		return message.guild!.security.actions.ban({
+			user_id: target.id,
+			moderator_id: message.author.id,
+			duration,
+			reason
+		}, Number(message.flagArgs.day || message.flagArgs.days) || 0, this.getTargetDM(message, target));
```

Comment:
Just a tiny nitpick, this won't pickup `DAY` or `DAYS`

Reply:
What's your suggested solution?

Reply:
Will be addressed on #460 with another PR.